docs/writing-tests/making-a-testing-plan.md

```diff
@@ -0,0 +1,504 @@
+number of drawbacks. It makes the test suite more difficult to understand since
+readers have to mentally "unwind" the iteration to determine what is actually
+being verified. It also increases the risk of bugs in the tests. These bugs may
+not be obvious--they may not cause failures, and they may exercise fewer cases
+than intended.
+
+Generally speaking, such exhaustive approaches are unlikely to catch more bugs
+than a handful of carefully-chosen test cases. Although the risks of dynamic
+test generation may be tolerable in some specific cases, it's usually best to
+select the most interesting edge cases and move on.
+
+*Example* We can see this consideration in the very first step of the
+`Response` constructor from [the Fetch
+standard](https://fetch.spec.whatwg.org/)
+
+> The `Response`(*body*, *init*) constructor, when invoked, must run these
+> steps:
+>
+> 1. If *init*["`status`"] is not in the range `200` to `599`, inclusive, then
+>    [throw](https://heycam.github.io/webidl/#dfn-throw) a `RangeError`.
+>
+> [...]
+
+This function accepts exactly 400 values for the "status." With [WPT's
+testharness.js](./testharness), it's easy to dynamically create one test for
+each value. Unless we have reason to believe that a browser may exhibit
+drastically different behavior for any of those values (e.g. correctly
+accepting `546` but incorrectly rejecting `547`), then the complexity of
+testing those cases probably isn't warranted.
+
+Instead, focus on writing declarative tests for specific values which are novel
+in the context of the algorithm. For ranges like in this example, testing the
+boundaries is a good idea. `200` and `599` should not produce an error while
+`199` and `600` should produce an error. Feel free to use what you know about
+the feature to choose additional values. In this case, HTTP response status
+codes are classified by the "hundred" order of magnitude, so we might also want
+to test a "3xx" value and a "4xx" value.
+
+## Assessing coverage
+
+It's very likely that WPT already has some tests for the feature (or at least
+the specification) that you're interesting in testing. In that case, you'll
+have to study what's already been done before starting to write new tests.
+Understanding the design of existing tests will let you avoid duplicating
+effort, and it will also help you integrate your work more logically.
+
+Even if the feature you're testing does *not* have any tests, you should still
+keep these guidelines in mind. Sooner or later, someone else will want to
+extend your work, so you ought to give them a good starting point!
+
+### File names
+
+The names of existing files and folders in the repository can help you find
+tests that are relevant to your work. [This page on the design of
+WPT](../test-suite-design) goes into detail about how files are generally laid
+out in the repository.
+
+Generally speaking, every conformance tests is stored in a subdirectory
```

Comment:
Is there some other documentation we can link to here? Something about "shortname" should be... somewhere.

Reply:
The W3C document is obsolete. [The link trail to a current version is a dead-end.](https://www.w3.org/pubrules) I think [this](https://www.w3.org/pubrules/doc) might be the latest, but I can't find a definition of "shortname" there.

Reply:
Linking to https://web-platform-tests.org/test-suite-design.html#test-layout might be best.

Reply:
That document is referenced by the preceding paragraph; this one is intended as a summary. Sounds like we're good!